### DISCUSSION TITLE
Sistema de colecionáveis no estilo NFTs?

### DISCUSSION THREAD

**Discussion by claudiolibanez** [diego3g/rsxp-2023#3] [Ideas]
Pensei algo como uma galeria de imagens (talvez uma imagem para cada atração assistida ou algo assim),  inicialmente que são desbloqueadas durante o evento através de um QR code. No futuro podem ser usadas como colecionáveis em outras plataformas ou coisa assim. O que acham?

**Comment by eletroswing** [comment]
Poderia ser um qr code com um link magnetico, ou gerando um código temporário, para evitar que pessoas compartilhem o qr code sem ter assistido ao evento!

**Reply by claudiolibanez** [reply]
Eu pensei em algo como um sticker com esse efeito:

https://youtu.be/pVesCl7TY8A

traria um plus bem diferente pro projeto

**Reply by eletroswing** [reply]
> Eu pensei em algo como um sticker com esse efeito:
> 
> https://youtu.be/pVesCl7TY8A
> 
> traria um plus bem diferente pro projeto

Realmente seria uma ótima forma de visualizar os stickers! Poderia ter um álbum por padrão e ao clicar, a gente teria um pop-up com a figurinha! E, seguindo a [discussão 5](https://github.com/diego3g/rsxp-2023/discussions/5), poderia existir um jeito de compartilhar seu progresso com o album, ou algo parecido!

**Reply by claudiolibanez** [reply]
mas como funcionaria o desbloqueio do sticker no backend?

Eu acho que consigo fazer um desenho do Diego em SD mandando um joinha e escrito "Fala Dev!" pra gente poder testar

**Reply by eletroswing** [reply]
a gente pode ter um token com expiração de 5 minutos, ai o usuário entra no app e valida o token.
Poderia ser também um deeplink chamando direto o app com o identificador da figurinha.
Ou a mistura de ambos?

o único limite é o que nós podemos pensar e fazer juntos.

**Reply by claudiolibanez** [reply]
Vamos fazer!

Se der certo eu faço os stickers no illustrator.

**Reply by ruymon** [reply]
Posso dar um help com UI! ;) 

Entretanto uma dúvida que precisamos levantar, será que para o MVP isso não é too much? 👨‍💻

![GIF](https://media1.giphy.com/media/IplvQCq2ZdwPUGcaVN/giphy.gif)

**Reply by claudiolibanez** [reply]
Acredito que o processo seria já existir uma pagina no app a galeria de imagens, no back retornar uma lista de tokens numa rota profile e a gente teria que arranjar um jeito de insertar esse token numa interação de qr code, codigo ou geolocalização e hora. Demais seria um deeplink ou algo parecido.. o que acham? é muita coisa?

**Reply by eletroswing** [reply]
as figurinhas levam aos badges citados, poderiamos começar com ela, e o sistema levaria aos badges em futuras versões, justamente pra não ser uma feature muito pesada e demorada de se pensar, as figurinhas são relativamente fáceis, já os badges dão um nó na cabeça para pensar em cada um e como cada um funciona

**Comment by ruymon** [comment]
@claudiolibanez Que tal juntarmos essa ideia com essa aqui #8 ?

**Reply by claudiolibanez** [reply]
Acabei de falar isso hahahaha

**Reply by ruymon** [reply]
Fecho!

**Comment by ruymon** [comment]
Discussão #8 Movida para cá. @claudiolibanez :)


---

# Insígnias e Medalhas 🎖
### Gamificando o app da RSXP 2023

**🧠 Ideia:** Ter medalhas ou badges para conquistas do pessoal durante toda a experiência do evento. Tal ideia se assemelha muito com o sistema atual na plataforma da @rocketseat.

![image](https://user-images.githubusercontent.com/50504050/228389645-d8775539-daee-4db9-bd33-f107c4328a07.png)

**🔍 Origem da ideia:** Essa ideia é uma adaptação do *insight* que o @diego3g teve durante a [live de kick-off](https://www.youtube.com/watch?v=dDO0YKuHABY). Ele citou a experiência que teve durante um evento, onde recebeu um álbum de figurinhas que foi completando com adesivos que recebeu pelos palestrantes. Com base nisso, pensei em uma maneira de integrar essa ideia de uma maneira digital e que consiga também agregar valor para o App e, por conseguinte, gamificar.

**🗒 Como funciona?** Ao realizar alguma ação, seja ela no evento ou fora dele, o usuário pode ganhar uma medalha e acumular conquistas. Nesse primeiro momento antes do evento, essa função pode estimular venda de ingressos.


**🎖 Ideias de Insígnias:**

- **Early bird |** Dada para usuários da v1 BETA do app.
- **Contributor |** Dada para pessoas que ajudaram no processo de construção do app.
- *Sales related |* Categoria de badges relacionadas à venda de ingressos. Ou seja, reconhecer o usuário que consegue vender `x` quantidade de ingressos.
- *Talk/Keynote related |* Categoria de badges para Talks, ou seja, recebe por comparecer na talk.
- **Hackathon Master! |** Dada aos vencedores da Hackathon.

Enfim... o céu é o limite para essas badges...

**💰 Como isso agrega valor ao app?** Essa feature estimula o uso do app antes, durante e depois do evento e, cria *hype* para o usuário pegar mais medalhas. Dessa forma promove com que ele curta 100% do evento. Além disso, abre a possibilidade de criar algum tipo de "lembrancinha" com as memórias pós RSXP...

**Reply by ruymon** [reply]
Outra inspiração é o sistema de *achievements* do @Github! 

![image](https://user-images.githubusercontent.com/50504050/228393046-1f1abd71-84bf-4765-8b78-f75a71037e3f.png)


[Todas as medalhas disponíveis.](https://github.com/Schweinepriester/github-profile-achievements)

**Reply by FelipeQueiroz** [reply]
Acredito que como primeira forma de interação para obter esses stickers pré evento seria de acordo com as indicações, como citei neste tópico https://github.com/diego3g/rsxp-2023/discussions/5#discussion-5015198

Dai para simplificar bastaria criar uma aba de "Colecionáveis" para usuário ter acesso a todas essas figurinhas :)

**Reply by eletroswing** [reply]
@FelipeQueiroz  @ruymon vamos separar, existiriam vários colecionáveis, como várias figurinhas e varias insígnias, então deveriamos ter uma aba "colecionáveis", e nela, poderia ter as insignias(mais raras e difíceis de se obter) e depois o álbum de figurinhas, o que acham?

**Reply by ruymon** [reply]
> @FelipeQueiroz @ruymon vamos separar, existiriam vários colecionáveis, como várias figurinhas e varias insígnias, então deveriamos ter uma aba "colecionáveis", e nela, poderia ter as insignias(mais raras e difíceis de se obter) e depois o álbum de figurinhas, o que acham?

Acho interessante, entretanto acho que complica muito. Entretanto é algo muito legal, confesso haha 😃. Caso sobre tempo durante a produção do app podemos fazer essas diferentes categorias.
